### PR TITLE
This PR implements color mapping variable for IsoSurface. 

### PR DESCRIPTION
**Pull request by shaomeng**

### CODE REVIEW

File: lib/render/RayCaster.cpp [hunk 1]
```diff
@@ -407,12 +455,118 @@ RayCaster::UserCoordinates::UpdateFaceAndData( const RayCasterParams* params,
...
+    dataFieldUpToDate = true;
+
     return 0;
 }
 
-void RayCaster::UserCoordinates::FillCoordsXYPlane( const  StructuredGrid* grid,
-                                                    size_t planeIdx,
-                                                    float* coords )
+int  
+RayCaster::UserCoordinates::Update2ndVariable( const RayCasterParams* params,
```
Comment (clyne) [inline lib/render/RayCaster.cpp]: There is a lot of code here cut-n-paste from RayCaster::UserCoordinates::UpdateFaceAndData. Please refactor to use helper functions and eliminate the code/functionality duplication.

File: lib/render/RayCaster.cpp [hunk 2]
```diff
@@ -745,6 +927,16 @@ int RayCaster::_initializeFramebufferTextures()
     glTexParameteri(GL_TEXTURE_3D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
     glTexParameteri(GL_TEXTURE_3D, GL_TEXTURE_WRAP_R, GL_CLAMP_TO_EDGE);
 
+    /* Generate and configure 3D texture: _2ndVarDataTexId */
+    glGenTextures( 1, &_2ndVarDataTexId );
```
Comment (StasJ) [inline lib/render/RayCaster.cpp]: All these textures have the same parameters so you could consolidate their initialization

File: lib/render/RayCaster.cpp (hunk 1 above)
Comment (shaomeng) [reply lib/render/RayCaster.cpp]: Addressed by commit 625e725

File: lib/render/DVRenderer.cpp [hunk 3]
```diff
@@ -41,9 +41,28 @@ int DVRenderer::_load3rdPassShaders()
...
+{
+    // Get colormap for the primary variable
+    DVRParams*             params     = dynamic_cast<DVRParams*>( GetActiveParams() );
+    VAPoR::MapperFunction* mapperFunc = params->GetMapperFunc();
+    mapperFunc->makeLut( _colorMap );
+    assert( _colorMap.size()  % 4 == 0 );
+    std::vector<double> range = mapperFunc->getMinMaxMapValue();
+    _colorMapRange[0]         = float(range[0]);
+    _colorMapRange[1]         = float(range[1]);
+    _colorMapRange[2]         = (_colorMapRange[1] - _colorMapRange[0]) > 1e-5f ?
```
Comment (clyne) [inline lib/render/DVRenderer.cpp]: Why is this needed? Please comment

File: lib/render/RayCaster.cpp (hunk 2 above)
Comment (shaomeng) [reply lib/render/RayCaster.cpp]: This is addressed in eedfe73.

File: lib/render/DVRenderer.cpp (hunk 3 above)
Comment (shaomeng) [reply lib/render/DVRenderer.cpp]: addressed in 6ac5e24

**Comment by shaomeng** [comment]
@StasJ @clyne : Please see my recent commits that address your comments.
